### ISSUE TITLE
Lookahead optimizer

### ISSUE THREAD

**Issue by jinmel** [tensorflow/tensorflow#31603]
**System information**
- TensorFlow version (you are using): 1.14
- Are you willing to contribute it (Yes/No):Yes

**Describe the feature and the current behavior/state.**

https://arxiv.org/abs/1907.08610

This is not a conventional optimizer as in tensorflow API. It observes through k iterations of optimizers(SGD, Adam), and linear interpolates entire weight into direction where fast weight after k iteration has reached. 

I have seen an implementation of this defining two ops for slow weight update and fast weight update. However I see some potential performance benefit if it was defined on backend. 

**Will this change the current api? How?**

I believe not.

**Who will benefit with this feature?**

Tasks with heavy training load will benefit from faster learning rate.

**Any Other info.**

**Comment by tanzhenyu** [comment]
Yep sounds good. @jinmel do you want to contribute to it?

**Comment by jinmel** [comment]
@tanzhenyu Sure

**Comment by jinmel** [comment]
preflight:
https://github.com/jinmel/lookahead-tensorflow

**Comment by nova77** [comment]
@jinmel thanks for your implementation, but it seems to be missing the sparse gradients update. Is this a limitation of this specific optimizer?

**Comment by jinmel** [comment]
@nova77 I still need to implement this feature.

**Comment by dynamicwebpaige** [comment]
This optimizer might be a better fit in [TensorFlow Addons](https://www.github.com/tensorflow/addons), then graduated to TensorFlow Core. @jinmel, do you have an update on progress for the sparse gradients update?

**Comment by WindQAQ** [comment]
Hi all, addons holds a lookahead optimizer implementation.
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/lookahead.py

**Comment by dynamicwebpaige** [comment]
Closing this issue, as there is an implementation available in TensorFlow Addons. Thank you for the feature request!
